api/common.go

```diff
@@ -27,12 +27,26 @@ const (
 	DefaultDockerfileName string = "Dockerfile"
 )
 
-// byPrivatePort is temporary type used to sort types.Port by PrivatePort
-type byPrivatePort []types.Port
+// portOrdering is a temporary type used to sort types.Port by its fields
+type portOrdering []types.Port
```

Comment:
Its not really alphabetical - its numeric and alphabetical depending on how deep it goes into the structure.
I don't really have an opinion on the name - other than it might be nice to start with "by" since that seems to be what most people start sort structs with.  But no biggie either way.  Perhaps: `byPortInfo`  ??